### PR TITLE
metric and top level risk functions

### PR DESCRIPTION
**Pull request by benholtzmansmith**
This PR addresses issue #<insert number>

### Please Check
- [ ] Are the [JSON schemas](https://github.com/covid-projections/covid-data-model/tree/master/api/schemas) updated?
- [ ] Are the [api docs](https://github.com/covid-projections/covid-data-model/blob/master/api/README.V1.md) updated?
- [ ] Are tests passing?

### CODE REVIEW

File: libs/top_level_metric_risk_levels.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        risk_level = RiskLevel.MEDIUM
+    elif value <= level_high:
+        risk_level = RiskLevel.HIGH
+    elif value > level_high:
+        risk_level = RiskLevel.CRITICAL
+    return risk_level
+
+
+def case_density_risk_level(value: float):
+    thresholds = [1, 10, 25]
```
Comment (benholtzmansmith) [inline libs/top_level_metric_risk_levels.py:42]: https://github.com/covid-projections/covid-projections/blob/1c15ef642eaa1247baa913c646fdf22590d0bf03/src/common/metrics/case_density.tsx#L22

File: libs/top_level_metric_risk_levels.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    return calc_risk_level(value, thresholds)
+
+
+def test_positivity_risk_level(value: float):
+    thresholds = [0.03, 0.1, 0.2]
```
Comment (benholtzmansmith) [inline libs/top_level_metric_risk_levels.py:47]: https://github.com/covid-projections/covid-projections/blob/1c15ef642eaa1247baa913c646fdf22590d0bf03/src/common/metrics/positive_rate.tsx#L33

File: libs/top_level_metric_risk_levels.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    return calc_risk_level(value, thresholds)
+
+
+def icu_headroom_ratio_risk_level(value: float):
+    thresholds = [0.5, 0.6, 0.7]
+    return calc_risk_level(value, thresholds)
+
+
+def infection_rate_risk_level(value: float):
+    thresholds = [0.9, 1.1, 1.4]
```
Comment (benholtzmansmith) [inline libs/top_level_metric_risk_levels.py:65]: https://github.com/covid-projections/covid-projections/blob/1c15ef642eaa1247baa913c646fdf22590d0bf03/src/common/metrics/case_growth.tsx#L27

File: libs/top_level_metric_risk_levels.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+def contact_tracing_risk_level(value: float):
+    # NOTE: Setting the upperLimit to 0 means we will not grade anybody as
+    # RiskLevel.LOW ("Critical" on the website). The lowest grade you can get is
+    # RiskLevel.MEDIUM.
+    thresholds = [0, 0.1, 0.9]
+    return calc_risk_level(value, thresholds)
+
+
+def icu_headroom_ratio_risk_level(value: float):
+    thresholds = [0.5, 0.6, 0.7]
```
Comment (benholtzmansmith) [inline libs/top_level_metric_risk_levels.py:60]: https://github.com/covid-projections/covid-projections/blob/1c15ef642eaa1247baa913c646fdf22590d0bf03/src/common/metrics/hospitalizations.tsx#L30-L32

File: libs/top_level_metric_risk_levels.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+def test_positivity_risk_level(value: float):
+    thresholds = [0.03, 0.1, 0.2]
+    return calc_risk_level(value, thresholds)
+
+
+def contact_tracing_risk_level(value: float):
+    # NOTE: Setting the upperLimit to 0 means we will not grade anybody as
+    # RiskLevel.LOW ("Critical" on the website). The lowest grade you can get is
+    # RiskLevel.MEDIUM.
+    thresholds = [0, 0.1, 0.9]
```
Comment (benholtzmansmith) [inline libs/top_level_metric_risk_levels.py:55]: https://github.com/covid-projections/covid-projections/blob/1c15ef642eaa1247baa913c646fdf22590d0bf03/src/common/metrics/contact_tracing.tsx#L38

File: libs/top_level_metric_risk_levels.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    thresholds = [0.5, 0.6, 0.7]
+    return calc_risk_level(value, thresholds)
+
+
+def infection_rate_risk_level(value: float):
+    thresholds = [0.9, 1.1, 1.4]
+    return calc_risk_level(value, thresholds)
+
+
+def top_level_risk_level(
```
Comment (benholtzmansmith) [inline libs/top_level_metric_risk_levels.py:69]: https://github.com/covid-projections/covid-projections/blob/1c15ef642eaa1247baa913c646fdf22590d0bf03/src/common/models/Projections.ts#L147

File: libs/top_level_metric_risk_levels.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    case_density_level: RiskLevel,
+    test_positivity_level: RiskLevel,
+    contact_tracing_level: RiskLevel,
+    icu_headroom_level: RiskLevel,
+    infection_rate_level: RiskLevel,
+):
+    """
+    Calculate the overall risk for a region based on metric risk levels.
+    """
+    levelList = [
```
Comment (ghop02) [inline libs/top_level_metric_risk_levels.py:79]: `level_list` for consistency

File: libs/top_level_metric_risk_levels.py [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        infection_rate_level,
+        icu_headroom_level,
+        test_positivity_level,
+        case_density_level,
+    ]
+    top_level_risk = RiskLevel.UNKNOWN
+
```
Comment (ghop02) [inline libs/top_level_metric_risk_levels.py:86]: I think you need to check if case_density is low first.  If case density is low we count the overall risk level as low

**Review by ghop02 — APPROVED** [review]
Couple small comments, but this is a great start

File: libs/top_level_metric_risk_levels.py [hunk 9]
```diff
@@ -0,0 +1,97 @@
+import enum
+from typing import List
+
+from covidactnow.datapublic import common_fields
+
+
+class RiskLevel(common_fields.ValueAsStrMixin, str, enum.Enum):
```
Comment (ghop02) [inline libs/top_level_metric_risk_levels.py:7]: I don't have a huge opinion, but I wonder if it'd be better to encode the enums as an int.  I'm just thinking about when we're writing this as a timeseries, probably don't need to have the string values

File: libs/dataset_deployer.py [hunk 10]
```diff
@@ -97,6 +98,10 @@ def write_nested_csv(
             # value for a, (i.e. {a: None}), don't write a, as it's not in the header.
             flattened_row = {k: v for k, v in flattened_row.items() if k in header}
             flattened_row = {k: v for k, v in flattened_row.items() if not pd.isnull(v)}
+            flattened_row = {
+                k: v.value if isinstance(v, enum.Enum) else v for k, v in flattened_row.items()
```
Comment (TomGoBravo) [inline libs/dataset_deployer.py:102]: maybe replace `k: v.value if ...` with `k: v` if https://github.com/covid-projections/covid-data-public/blob/master/covidactnow/datapublic/common_fields.py#L20 on the enum class gets writerow to do what you want.

File: libs/top_level_metric_risk_levels.py (hunk 7 above)
Comment (TomGoBravo) [reply libs/top_level_metric_risk_levels.py:87]: I don't see this fixed.

**Review by TomGoBravo — APPROVED** [review]
It'd be nice to have this calculated with fast panda operations but get that what we have here works and is easier to read.

**Pull request merged**